Classes/Search/GithubClient+Search.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  GithubClient+Search.swift
+//  Freetime
+//
+//  Created by Sherlock, James on 28/07/2017.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import Foundation
+
+extension GithubClient {
+    
+    enum SearchResultType {
+        case error
+        case success(String?, [SearchResult])
+    }
+    
+    func search(query: String, before: String? = nil, containerWidth: CGFloat, completion: @escaping (SearchResultType) -> ()) {
+        let query = SearchReposQuery(search: query, before: before)
+        
+        apollo.fetch(query: query, cachePolicy: .fetchIgnoringCacheData) { (result, error) in
+            guard error == nil, result?.errors == nil else {
+                ShowErrorStatusBar(graphQLErrors: result?.errors, networkError: error)
+                return
```

Comment:
I have seen these a couple of times in this PR (a `return` when the method accepts a completion handler). As a developer interactive with this API I would always expect my completion block to be called? 🤔  Is there something I am missing here that this is maybe intentional?